cli/root.go

```diff
@@ -137,6 +142,17 @@ func Execute() {
 	app.Flag("timeout", "Timeout for the executed command").Default("30s").DurationVar(&timeout)
 	app.Flag("http.config.file", "HTTP client configuration file for amtool to connect to Alertmanager.").PlaceHolder("<filename>").ExistingFileVar(&httpConfigFile)
 	app.Flag("version-check", "Check alertmanager version. Use --no-version-check to disable.").Default("true").BoolVar(&versionCheck)
+	app.Flag("enable-feature", fmt.Sprintf("Experimental features to enable. The flag can be repeated to enable multiple features. Valid options: %s", strings.Join(featurecontrol.AllowedFlags, ", "))).Default("").StringVar(&featureFlags)
+
+	logger := log.NewLogfmtLogger(os.Stdout)
+	featureConfig, err := featurecontrol.NewFlags(logger, featureFlags)
+	if err != nil {
+		kingpin.Fatalf(":error parsing the feature flag list: %v\n", err)
+	}
+	if featureConfig.DisableNewLabelMatchers() {
+		adapter.ParseMatcher = adapter.OldMatcherParser(logger)
+		adapter.ParseMatchers = adapter.OldMatchersParser(logger)
```

Comment:
I think I would prefer if we made the compact package stateful that way we wouldn't need to do global variables.

```golang
compatibleParser := compat.NewParser(featureConfig) 
```

Then callers can do: 

```golang
compatibleParser.ParseMatcher()
```

Reply:
I was thinking here in `root.go` but perhaps there is a problem with that?

Reply:
Yes `root.go` is what I meant 😅 oops!

But if `compatibleParser` is a local variable in `root.go`, how do we propagate it to [`utils.go`](https://github.com/prometheus/alertmanager/blob/main/cli/utils.go#L41) for example?